File: Programowanie_Obiektowe/PO_JAVA/src/Zolw.java
public class Zolw extends Zwierze
{
    public Zolw(int x, int y, Swiat swiat)
    {
        super(Stale.ZOLW_SILA, Stale.ZOLW_INICJATYWA, x, y, swiat);
    }

    @Override
    public void akcja()
    {
        wiek++;

        if (new java.util.Random().nextInt(4) == 0)     // 25% szansy na ruch
        {
            int[] XY = swiat.wylosujSasiada(this);
            if (XY != null)
            {
                przesun(XY[0], XY[1]);
            }
            else
            {
                swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") brak dostepnych ruchow");
            }
        }
        else
        {
            swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") nie rusza się");
        }
    }

    @Override
    public void kolizja(Organizm atakujacy)
    {
        if (this.getClass() == atakujacy.getClass())
        {
            rozmnazanie();
        }
        else
        {
            if (atakujacy.getSila() < Stale.ZOLW_ODPIERA)
            {
                swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") odparł atak " + atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ", " + atakujacy.getSila() + ")");
            }
            else
            {
                atak(atakujacy);
            }
        }
    }

    @Override
    public char rysowanie()
    {
        return Stale.ZOLW;
    }

    @Override
    public String nazwa()
    {
        return Stale.ZOLW_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Zolw(x, y, swiat);
    }
}

File: Programowanie_Obiektowe/PO_C/PO_C/Zolw.cpp
#include "Zolw.h"

Zolw::Zolw(int x, int y, Swiat& swiat) : Zwierze(ZOLW_SILA, ZOLW_INICJATYWA, x, y, swiat) {}

void Zolw::akcja()
{
    wiek++;

    if (rand() % 4 == 0)    //25% na ruch
    {
        int dx[8] = { -1, 0, 1, -1, 1, -1, 0, 1 };
        int dy[8] = { 1, 1, 1, 0, 0, -1, -1, -1 };

        while (true)
        {
            int dir = rand() % 8;

            int newX = x + dx[dir];
            int newY = y + dy[dir];

            if (swiat.czyNaPlanszy(newX, newY))
            {
                przesun(newX, newY);
                break;
            }
        }
    }
    else
    {
        swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ", " + to_string(sila) + ") nie rusza sie " + "\n");
    }
}

void Zolw::kolizja(Organizm& atakujacy)
{
    if (typeid(*this) == typeid(atakujacy))          //rozmnarzanie
    {
        rozmnazanie();
    }
    else
    {
        if (atakujacy.getSila() < ZOLW_ODPIERA)     //zolw odpiera atak
        {
            swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ", " + to_string(sila) + ") odparl atak " + string(1, atakujacy.rysowanie()) + " (" + to_string(atakujacy.getX()) + ", " + to_string(atakujacy.getY()) + ", " + to_string(atakujacy.getSila()) + ")" + "\n");
        }
        else
        {
            atak(atakujacy);
        }
    }
}

char Zolw::rysowanie() const
{
    return ZOLW;
}

Organizm* Zolw::stworzPotomka(int x, int y) const
{
    return new Zolw(x, y, swiat);
}